build-system/compile/compile.js

```diff
@@ -236,15 +236,27 @@ function compile(
     }
     externs.push('build-system/externs/amp.multipass.extern.js');
 
+    // Normally setting this server-side experiment flag would be handled by
+    // the release process automatically. Since this experiment is actually on the
+    // build system instead of runtime, we never run it through babel and therefore
+    // must compute it here.
+    const STRICT_COMPILATION =
+      argv.define_experiment_constant &&
+      argv.define_experiment_constant === 'STRICT_COMPILATION';
+    let target = STRICT_COMPILATION ? 'ECMASCRIPT5_STRICT' : 'ECMASCRIPT5';
+    if (argv.esm) {
+      // Do not transpile down to ES5 if running with `--esm`, since we do
+      // limited transpilation in Babel.
+      target = 'NO_TRANSPILE';
+    }
```

Comment:
Thanks for the explanation. I now see how this is different from the other experiments. I've made a stylistic suggestion to reorganize this code below. (Renamed the all-caps `STRICT_COMPILATION` constant to `isStrict`, renamed `target` to `language` since it's being passed to `language_out`, and simplified the language selection logic.) Feel free to accept, decline, or modify as you please.

In future, we should refactor this function to be less huge than it currently is 😃 

```suggestion
    // Normally, setting this server-side experiment flag would be handled by
    // the release process automatically. Since this experiment is actually on the
    // build system instead of runtime, we never run it through babel and therefore
    // must compute it here.
    const isStrict =
      argv.define_experiment_constant &&
      argv.define_experiment_constant === 'STRICT_COMPILATION';
    // Do not transpile down to ES5 if running with `--esm`, since we do
    // limited transpilation in Babel.
    const isEsm = argv.esm;
    const language = isEsm
      ? 'NO_TRANSPILE'
      : isStrict
      ? 'ECMASCRIPT5_STRICT'
      : 'ECMASCRIPT5';
```

Reply:
I personally try to avoid nested ternaries :).  I'll take most of these suggestions though and make it more like:

```js
const isStrict = ...;
const isEsm = argv.esm;
let language;
if (isEsm) {
  ...
} else if (isStrict) {
  ...
} else {
  ...
}
```